pkg/config/config.go

```diff
@@ -74,15 +74,58 @@ func LoadConfigFromString(configString string) (*Config, error) {
 		return nil, fmt.Errorf("failed to unmarshal YAML file: %w", err)
 	}
 
-	// TODO: We should split these up and add error messages
-	if config.Matrix.UserID == "" ||
-		config.Matrix.HomeserverURL == "" ||
-		config.Matrix.AccessToken == "" ||
-		config.Conference.KeepAliveTimeout == 0 ||
-		config.Conference.KeepAliveTimeout > 30 ||
-		config.Conference.PingInterval < 30 {
-		return nil, errors.New("invalid config values")
+	if err := ValidateConfig(config); err != nil {
+		return nil, err
 	}
 
 	return &config, nil
 }
+
+func ValidateConfig(config Config) error {
+	errored := false
+
+	if config.Matrix.UserID == "" {
+		errored = true
+		fmt.Println("You must set matrix.userId")
+	}
+	if config.Matrix.HomeserverURL == "" {
+		errored = true
+		fmt.Println("You must set matrix.homeserverUrl")
+	}
+	if config.Matrix.AccessToken == "" {
+		errored = true
+		fmt.Println("You must set matrix.accessToken")
+	}
+	if config.Conference.PingPongConfig.Timeout == 0 {
+		errored = true
+		fmt.Println("You must set pingPong.timeout")
+	}
+	if config.Conference.PingPongConfig.Interval == 0 {
+		errored = true
+		fmt.Println("You must set pingPong.interval")
+	}
+
+	if config.Conference.PingPongConfig.Timeout > 30 {
+		errored = true
+		fmt.Println("pingPong.timeout must be 30s or lower")
+	}
+	if config.Conference.PingPongConfig.Interval < 30 {
+		errored = true
+		fmt.Println("pingPong.interval must be 30s or higher")
+	}
+
+	if config.Conference.PingPongConfig.Timeout < 5 {
+		errored = true
+		fmt.Println("It is not recommended for pingPong.timeout to be below 5s")
+	}
+	if config.Conference.PingPongConfig.Interval > 60*5 {
+		errored = true
+		fmt.Println("It is not recommended for pingPong.interval to be more than 300s")
+	}
+
+	if errored {
+		return errors.New("invalid config values")
+	} else {
+		return nil
+	}
+}
```

Comment:
Generally `fmt.Println()` is not the best way to report errors when doing it outside of a `main` package (also it uses the stdout and may collide with whatever logging framework is used). For packages and libraries, it's more common to just return an error (the caller will unpack the error and print it if it's of significance).

I think the function will look more elegant and concise if each `if` returns a certain error. The caller (in `main()`) would anyway get this error and print it via the logging framework.

Reply:
One option would be to create a couple of errors in the top of the file and them parameterize them with the field that failed or something like this, but ok, let me approve and improve it later!